src/main/java/edu/cornell/kfs/module/purap/batch/dataaccess/impl/JaggaerUploadDaoJdbc.java

```diff
@@ -100,92 +106,100 @@ public List<JaggaerContractAddressUploadDto> findJaggaerContractAddress(JaggaerC
-        sb.append("AND VA.VNDR_ADDR_TYP_CD in ('RM', 'PO') ");
-        sb.append(buildOrderByClause());
-        
-        String sql = sb.toString();
-        LOG.info("buildVendorAddressSql, SQL: " + sql);
-        return sql;
+    private CuSqlChunk buildVendorAddressSqlCuSqlChunk(JaggaerContractUploadProcessingMode processingMode, String processingDate) {
+        Collection<String> addressTypes = Arrays.asList(CUPurapConstants.JaggaerAddressType.REMIT.kfsAddressTypeCode, 
+                CUPurapConstants.JaggaerAddressType.FULFILLMENT.kfsAddressTypeCode);
+        CuSqlChunk chunk = CuSqlChunk.of("SELECT VA.VNDR_ADDR_GNRTD_ID, VA.VNDR_ADDR_TYP_CD, VA.VNDR_DFLT_ADDR_IND, VA.VNDR_CNTRY_CD, VA.VNDR_LN1_ADDR, ",
+                "VA.VNDR_LN2_ADDR, VA.VNDR_CTY_NM, VA.VNDR_ST_CD, VA.VNDR_ADDR_INTL_PROV_NM, VA.VNDR_ZIP_CD, VA.VNDR_HDR_GNRTD_ID, VA.VNDR_DTL_ASND_ID ",
+                buildFromClauseCuSqlChunk(true),
+                buildJoinClauseWithPOVendorsCuSqlChunk(true),
+                buildRestrictActiveRowsCuSqlChunk(true),
+                buildTimeFrameRestrictionClauseCuSqlChunk(processingMode, processingDate),
+                "AND VA.VNDR_ADDR_TYP_CD in (",
+                CuSqlChunk.forStringParameters(addressTypes),
+                ") ",
+                buildOrderByClauseCuSqlChunk());
+        return chunk;
     }
     
-    private String buildFromClause(boolean includeVendorAddress) {
-        StringBuilder sb = new StringBuilder("FROM KFS.PUR_VNDR_HDR_T VH, KFS.PUR_VNDR_DTL_T VD");
+    private CuSqlChunk buildFromClauseCuSqlChunk(boolean includeVendorAddress) {
+        CuSqlChunk chunk = CuSqlChunk.of("FROM KFS.PUR_VNDR_HDR_T VH, KFS.PUR_VNDR_DTL_T VD ");
         if (includeVendorAddress) {
-            sb.append(", KFS.PUR_VNDR_ADDR_T VA");
+            chunk.append(", KFS.PUR_VNDR_ADDR_T VA ");
         }
-        sb.append(StringUtils.SPACE);
-        return sb.toString();
+        return chunk;
     }
     
-    private String buildJoinClauseWithPOVendors(boolean includeVendorAddress) {
-        StringBuilder sb = new StringBuilder("WHERE VH.VNDR_HDR_GNRTD_ID = VD.VNDR_HDR_GNRTD_ID ");
+    private CuSqlChunk buildJoinClauseWithPOVendorsCuSqlChunk(boolean includeVendorAddress) {
+        CuSqlChunk chunk = CuSqlChunk.of("WHERE VH.VNDR_HDR_GNRTD_ID = VD.VNDR_HDR_GNRTD_ID ");
         if (includeVendorAddress) {
-            sb.append("AND VA.VNDR_HDR_GNRTD_ID = VD.VNDR_HDR_GNRTD_ID ");
-            sb.append("AND VA.VNDR_DTL_ASND_ID = VD.VNDR_DTL_ASND_ID ");
+            chunk.append("AND VA.VNDR_HDR_GNRTD_ID = VD.VNDR_HDR_GNRTD_ID ");
+            chunk.append("AND VA.VNDR_DTL_ASND_ID = VD.VNDR_DTL_ASND_ID ");
         }
-        sb.append("AND VH.VNDR_TYP_CD = 'PO' ");
-        return sb.toString();
+        chunk.append("AND VH.VNDR_TYP_CD =");
+        chunk.append(CuSqlChunk.forParameter(CUPurapConstants.JaggaerAddressType.FULFILLMENT.kfsAddressTypeCode));
+        chunk.append(StringUtils.SPACE);
```

Comment:
I have a few thoughts on how to help make sections like this one less verbose and easier to read:

* The CuSqlChunk class is set up so that you can chain multiple .append() calls together, like what Java's StringBuilder can do. That should help with grouping related sub-chunks together.
* You could modify the section-building areas to start with a leading space, rather than ending with a trailing space.
* Perhaps some new methods could be added to CuSqlChunk to simplify the SQL-building process (with appropriate unit tests added to CuSqlQueryTest). Here are some thoughts on new methods that might help:
  * `append(CharSequence... subChunks)` -- Behaves similarly to CuSqlChunk.of() except that it appends to the existing CuSqlChunk instance. (You might not need this one if you decide to implement the two below.)
  * `appendParameter(String value)` and `appendParameter(int sqlType, Object value)` -- Shorthand methods for appending a SQL parameter that automatically handle constructing the SqlParameterValue instance.

Reply:
adding a new append(CharSequence... subChunks) function seems to have cleaned. a lot up